### PR TITLE
Email applicant when membership is granted #153348015

### PR DESCRIPTION
**Pull request by weedySeaDragon**
## PT Story:  Send email to applicant when membership is granted
https://www.pivotaltracker.com/story/show/153348015

The text needs to be decided on.  I put in placeholders and text that was originally written for the wrong story (was written for the 'SHF membership application approved, but should have been for this story).

## Changes proposed in this pull request:
1.  Created a `MemberMailer`  to send emails out to members.  I decided to create this class instead of using the existing  UserMailer because (a) I knew that there would be information specific to a Member that would be needed (e.g. existing company, membership application), and the UserMailer is primarily used to send out Devise-related emails right now.
2. Refactored some Mailers methods and views because they are now used by multiple Mailer classes (ex: the `signoff...` and `email_to_membership...` partials)
3.  Needed to add a parameter to the `User .grant_membership` method so that we can specify that we do _not_ want to send out emails during seeding. (`db/seed_helpers.rb line 100`
3. Mocked the MemberMailer in User spec tests and PaymentsController spec tests  (used a `double`) 
4. Reorganized the locale files for the mailer entries

Tuesday 19 Dec: I updated the text per your suggestion & Bella's, @thesuss .  
Let me know if it's correct or not.

## Screenshots (Optional):

email-membership-granted-2-html

email-membership-granted-2-text

### Ready for review:
@AgileVentures/shf-project-team

### CODE REVIEW

**Comment by thesuss** [comment]
"Ditt företag är nu H-märkt och du har aktiverats som medlem i systemet"

The first two lines are (to me) unnecessary, the first sentence can change to include H-märkt as above, and then it should be fine.

**Review by thesuss — APPROVED** [review]

**Review by patmbolger — APPROVED** [review]

File: app/models/user.rb
Comment (patmbolger) [inline app/models/user.rb:107]: On second review I'd like to hold off on this PR for now - until we work out how and where we determine we will grant membership.  

Currently, this method is being called from the webhook that handles HIPS payments updates.  That webhook should operate as fast as possible - in order to complete required processing by the time HIPS sends control back to us - and sending an email might jeopardize that.

File: config/locales/sv.yml
Comment (thesuss) [inline config/locales/sv.yml]: När du är inloggad kan du redigera din företagssida och betala din H-märkesavgift här: 

(or is there something else being invoked to tell them to pay? )

Comment (weedySeaDragon) [reply config/locales/sv.yml]: At this point, they have already paid.  They get this email once they have satisfied _all of the requirements for membership_.  (So their application has been _approved_ and they have paid their membership fee.)

This email is separate from anything to do with their company.
(They have already received information about paying the company branding license fee (H-mäkt) [if needed] in an email they got after their application was approved.  See: 
https://github.com/AgileVentures/shf-project/wiki/Emails-automatically-sent-out

File: app/models/user.rb
Comment (weedySeaDragon) [reply app/models/user.rb:107]: Good point. Let's talk about this in Slack. :-)

Comment (patmbolger) [reply app/models/user.rb:107]: As discussed, I agree with going ahead with this PR.

**Comment by weedySeaDragon** [comment]
Per our conversation in Slack, I'll fix the text to say:

När detta är gjort behöver du göra färdigt din företagssida och betala din H-märkesavgift. Mer information om hur du gör det kommer i ett mail efter att du betalt din medlemsavgift.

**Review by RobertCram — APPROVED** [review]

File: config/locales/en.yml
Comment (thesuss) [inline config/locales/en.yml]: Translation, if you want:
When this is done you will need to finish your company page and pay your H-branding fee. More information about how to do that will come after you paid your membership fee.

**Pull request merged**
